test-framework/db-client/src/main/java/io/quarkus/test/db/client/DbTestClient.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.test.db.client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+
+import io.quarkus.datasource.runtime.DatabaseSchemaProvider;
+import io.quarkus.test.common.DevServicesContext;
+
+public class DbTestClient implements DevServicesContext.ContextAware {
+
+    private static final String DEFAULT_DB = "<default>";
+
+    private DevServicesContext testContext;
+    private final List<DatabaseSchemaProvider> providers = new ArrayList<>();
+
+    public DbTestClient() {
+    }
+
+    public void resetDefaultDatabase() {
+        resetDatabase(DEFAULT_DB);
+    }
+
+    public void resetDatabase(String dbName) {
+        for (DatabaseSchemaProvider i : providers) {
+            i.resetDatabase(dbName);
+        }
+    }
+
+    public void resetAllDatabases() {
+        for (DatabaseSchemaProvider i : providers) {
+            i.resetAllDatabases();
+        }
+    }
+
+    @Override
+    public void setIntegrationTestContext(DevServicesContext context) {
```

Comment:
It looks like this method is only called for integration tests, or on resource manager classes.

Reply:
@stuartwdouglas Thanks, yes, I was looking at the call chain for `setIntegrationTestContext` yesterday and it did show it was the case, and this is exactly why I don't understand how [this quickstart test](https://github.com/quarkusio/quarkus-quickstarts/blob/main/security-openid-connect-quickstart/src/test/java/org/acme/security/openid/connect/BearerTokenAuthenticationTest.java) works in JVM mode :-). The only difference is that there we also have a [native QuarkusIntegrationTest](https://github.com/quarkusio/quarkus-quickstarts/blob/main/security-openid-connect-quickstart/src/test/java/org/acme/security/openid/connect/NativeBearerTokenAuthenticationIT.java), so I wonder, can be the case that having `@QuarkusIntegrationTest` on the native test affects the initialization of the `@QuarkusTest`, may there is some magic going on there ?
Also CC @geoand

Reply:
> so I wonder, can be the case that having @QuarkusIntegrationTest on the native test affects the initialization of the @QuarkusTest, may there is some magic going on there ?

That shouldn't be happening

Reply:
It is handled manually by the resource manager: https://github.com/stuartwdouglas/quarkus/blob/cb6cf2d2af1d4feb00021a04d687e22d42a514db/integration-tests/oidc-token-propagation/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L102

Reply:
@stuartwdouglas yeah, but in https://github.com/quarkusio/quarkus-quickstarts/blob/main/security-openid-connect-quickstart/src/test/java/org/acme/security/openid/connect/BearerTokenAuthenticationTest.java no resource manager exists

Reply:
https://github.com/quarkusio/quarkus-quickstarts/tree/main/security-openid-connect-quickstart/src/test/java/org/acme/security/openid/connect

Reply:
Sorry, the reason KeycloakTestClient works is because it gets the properties it needs from ConfigProvider